app/models/release_platform_run.rb

```diff
@@ -236,4 +243,12 @@ def last_successful_run_for(step)
       .order(created_at: :desc)
       .first
   end
+
+  def previous_successful_run_before(step_run)
+    step_runs
+      .where(step: step_run.step)
+      .where.not(id: step_run.id)
+      .success
+      .last
+  end
 end
```

Comment:
can't we re-use the existing method `last_successful_run_for(step)`? the current step_run will never be in `success` status.